src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -635,6 +636,18 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 	MonoTypeEnum arg0_type = fsig->param_count > 0 ? get_underlying_type (fsig->params [0]) : MONO_TYPE_VOID;
 
 	switch (id) {
+	case SN_Abs: {
+#ifdef TARGET_ARM64
+        MonoType *arg_type = get_vector_t_elem_type (fsig->params [0]);
+		if (!MONO_TYPE_IS_INTRINSICS_VECTOR_PRIMITIVE (arg_type))
```

Comment:
I wanted to check if type is numeric but I see now that it doesn't check that. Do we have some function to understand if type is numeric?

Reply:
I had the impression that it is guaranteed that it should be numeric type, cause the other cases wasn't checking this.

Reply:
It validates that `T` is actually a supported type for `Vector64/128/256` as not every primitive type is supported.

Notably, it can be relaxed from https://github.com/dotnet/runtime/blob/04bb6eb3f182c934466af9c6f3d5a9ce8efdf716/src/mono/mono/mini/simd-intrinsics.c#L599
```
- #define MONO_TYPE_IS_INTRINSICS_VECTOR_PRIMITIVE(t) ((MONO_TYPE_IS_VECTOR_PRIMITIVE(t)) && ((t)->type != MONO_TYPE_I) && ((t)->type != MONO_TYPE_U))
+ #define MONO_TYPE_IS_INTRINSICS_VECTOR_PRIMITIVE(t) MONO_TYPE_IS_VECTOR_PRIMITIVE(t)
```

Where `MONO_TYPE_IS_VECTOR_PRIMITIVE` is: https://github.com/dotnet/runtime/blob/6c72db83cdab31786f602c782ded165c20d9eac8/src/mono/mono/mini/mini.h#L142

In .NET 6, `Vector64/128/256<T>` only support:
* `byte` and `sbyte`
* `short` and `ushort`
* `int` and `uint`
* `long` and `ulong`
* `float` and `double`

While `Vector<T>` supports those plus `nint` and `nuint`.
In .NET 7, `Vector64/128/256<T>` also support `nint` and `nuint` bringing them inline.

We do not support `bool`, `char`, `decimal`, or `half` today (or any of the other types that may be built-in or specially recognized).

Reply:
I see, thanks for the detailed explanation.

Reply:
@tannergooding So it is not guaranteed that runtime would only be given the supported types?

Reply:
Gotcha!

Reply:
Created an issue #65318 to track the cleanup work.